rewrite-java/src/main/java/org/openrewrite/java/tree/MethodTypeBuilder.java

```diff
@@ -36,32 +56,67 @@ public static MethodTypeBuilder methodType() {
         return new MethodTypeBuilder();
     }
 
+    /**
+     * @param fullyQualifiedClassName The fully-qualified name of the class on which this method is defined.
+     */
     public MethodTypeBuilder declaringClass(String fullyQualifiedClassName) {
         declaringType = JavaType.Class.build(fullyQualifiedClassName);
         return this;
     }
 
+    /**
+     * @param flags a list of qualifiers (public, static, etc) for the method
+     */
     public MethodTypeBuilder flags(Flag ... flags) {
         this.flags.addAll(Arrays.asList(flags));
         return this;
     }
 
+    /**
+     * @param name The name of the method is required
+     */
     public MethodTypeBuilder name(String name) {
         this.name = name;
         return this;
     }
 
+    /**
+     * @param type The return type for the method. The default, it no specified, is "void"
+     */
     public MethodTypeBuilder returnType(String type) {
         this.returnType = JavaType.buildType(type);
         return this;
     }
 
+    /**
+     * Add a parameter to the method type. Parameters are added in the order in which this method is called.
+     *
+     * The parameter type is either expressed as a fully-qualified class name or can be one of the primitive
+     * {@link JavaType.Primitive} keywords.
+     *
+     * @param type String representation of the parameter type.
+     * @param name The name of the parameter.
+     */
     public MethodTypeBuilder parameter(String type, String name) {
         this.parameters.add(new Parameter(JavaType.buildType(type), name));
         return this;
```

Comment:
Why does this even accept a `type` parameter when that information just gets thrown away later?

If there is a reason I'm just not seeing:
In many cases we'll already have a `JavaType` on hand or can easily construct one. I suggest that for this and `returnType` we accept `JavaType`s. `String` overload optional.

Reply:
Return type and parameter types are used to create the method signatures.